src/sonic-eventd/src/eventconsume.h

```diff
@@ -0,0 +1,54 @@
+#ifndef __EVENTCONSUME_H__
```

Comment:
I would like to see the existing event infra and this new changes work together in the unit test.

For example, if there is a BGP event, how does it get saved in the EVENT_DB.

Reply:
UT code added with this PR.